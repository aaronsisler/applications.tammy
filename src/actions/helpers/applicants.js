export const setApplicants = (applicants) => ({
    type: 'SET_APPLICANTS',
    applicants
});
