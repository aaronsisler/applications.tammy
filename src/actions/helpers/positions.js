export const setPositions = (positions) => ({
    type: 'SET_POSITIONS',
    positions
});
