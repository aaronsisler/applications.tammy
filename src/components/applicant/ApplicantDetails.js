import React from 'react';
import { connect } from 'react-redux';
import history from 'Tools/history';
import PropTypes from 'prop-types';
import ApplicantDetailsContent from 'Applicant/ApplicantDetailsContent';

export class ApplicantDetails extends React.Component {
    constructor(props) {
        super(props);
    }

    handleNavigateBack = () => history.push(`/applicants/${this.props.positionId}`);

    render() {
        const { applicant } = this.props;

        if (!applicant) {
            return (
                <div className="inbox_details__empty inbox_mobile">
                    Please select an applicant to view
                </div>
            )
        }
        return (
            <div className="inbox_details" >
                <div className="inbox_details_header">
                    <div className="inbox_details_header__actions">
                        <button
                            className="inbox_details_header__mobile_button"
                            onClick={this.handleNavigateBack}
                        >
                            Back to List
                        </button>
                    </div>
                    <div className="inbox_details_header__content">
                        <div className="inbox_details_header__display_name">
                            {applicant.user.displayName}
                        </div>
                        <div className="inbox_details_header__phone_number">
                            <strong>Phone:</strong>&nbsp;{applicant.user.displayPhoneNumber || "Not Provided"}
                        </div>
                        <div className="inbox_details_header__email">
                            <strong>Email:</strong>&nbsp;{applicant.user.email || "Not Provided"}
                        </div>
                    </div>
                </div>
                <div className="inbox_details_content">
                    <ApplicantDetailsContent applicant={applicant} />
                </div>
            </div>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = (state, props) => {
    const { applicantId, positionId } = props.match.params;

    const applicant = applicantId
        ? state.applicants.find((stateApplicant) => stateApplicant.applicantId == applicantId)
        : undefined;

    return ({
        applicant,
        applicantId,
        positionId,
    });
};

export default connect(mapStateToProps)(ApplicantDetails);

ApplicantDetails.propTypes = {
    applicant: PropTypes.object,
    applicantId: PropTypes.string,
    positionId: PropTypes.string.isRequired,
};

