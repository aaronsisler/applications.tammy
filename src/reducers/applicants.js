const applicantsReducerDefaultState = [];

const applicantsReducer = (state = applicantsReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_APPLICANTS':
            return action.applicants || applicantsReducerDefaultState;
        default:
            return state;
    }
};

export default applicantsReducer;

