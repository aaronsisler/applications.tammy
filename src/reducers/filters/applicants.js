const applicantsFilterReducerDefaultState =
{
    text: '',
};

const applicantsFilterReducer = (state = applicantsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_APPLICANTS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        default:
            return state;
    }
};

export default applicantsFilterReducer;
