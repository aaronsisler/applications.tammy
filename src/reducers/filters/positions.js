const positionsFilterReducerDefaultState =
{
    text: '',
};

const positionsFilterReducer = (state = positionsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'CLEAR_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: ''
            };
        default:
            return state;
    }
};

export default positionsFilterReducer;
