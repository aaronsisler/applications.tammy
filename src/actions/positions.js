import database from '../firebase/firebase';
import { setPositions } from 'Actions/helpers/positions';

export const startSetPositions = () => (dispatch) =>
    database.ref(`positions`).once('value').then((snapshot) => {
        const positions = [];
        snapshot.forEach((childSnapshot) => {
            positions.push({
                positionId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        dispatch(setPositions(positions));
    });
