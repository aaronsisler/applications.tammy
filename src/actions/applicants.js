import database from 'Firebase/firebase';
import { setApplicants } from 'Actions/helpers/applicants';
import { setApplicantsTextFilter } from 'Actions/filters/applicantsText';

export const startSetApplicants = () => (dispatch, getState) => {
    const { positionId } = getState().workflow.position;

    return database.ref(`applicants/${positionId}`).once('value').then((snapshot) => {
        const applicants = [];
        snapshot.forEach((childSnapshot) => {
            const applicant = childSnapshot.val();
            const applicantNotes = applicant.applicantNotes
                ? [...Object.values(applicant.applicantNotes)]
                : [];
            applicants.push({
                applicantId: childSnapshot.key,
                positionId,
                ...childSnapshot.val(),
                applicantNotes,
            });
        });

        return dispatch(setApplicants(applicants));
    });
}

export const startSetApplicantsTextFilter = (text) => (dispatch) => dispatch(setApplicantsTextFilter(text));
