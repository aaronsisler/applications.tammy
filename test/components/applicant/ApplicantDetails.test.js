import React from 'react';
import { shallow } from 'enzyme';
import { ApplicantDetails } from 'Applicant/ApplicantDetails';
import { applicant, applicantId } from '../../fixtures/applicants';
import { positionId } from '../../fixtures/positions';

import history from 'Tools/history';
jest.mock('Tools/history');

const buildWrapper = (applicantInput) => shallow(
    <ApplicantDetails
        applicant={applicantInput}
        applicantId={applicantId}
        positionId={positionId}
    />
);

describe('ApplicantDetails', () => {
    let wrapper;
    const push = jest.fn();

    describe('when applicant is defined', () => {
        describe('when user contact info is provided', () => {
            beforeEach(() => {
                wrapper = buildWrapper(applicant);
            });

            it('should render ApplicantUserInfo correctly', () => {
                expect(wrapper).toMatchSnapshot();
            });
        });

        describe('when user contact info is NOT provided', () => {
            beforeEach(() => {
                const applicantWithoutContactInto = {
                    ...applicant,
                    user: {
                        ...applicant.user,
                        displayPhoneNumber: undefined,
                        email: undefined
                    }
                }
                wrapper = buildWrapper(applicantWithoutContactInto);
            });

            it('should render ApplicantUserInfo correctly', () => {
                expect(wrapper).toMatchSnapshot();
            });
        });

        describe('when button is clicked', () => {
            beforeEach(() => {
                history.push = push
                wrapper = buildWrapper(applicant);
            });

            it('should navigate to correct route', () => {
                wrapper.find('button').simulate('click');

                expect(push).toHaveBeenLastCalledWith(`/applicants/${positionId}`);
            });
        });
    });

    describe('when applicant is undefined', () => {
        beforeEach(() => {
            wrapper = shallow(
                <ApplicantDetails
                    positionId={positionId}
                />
            );
        });

        it('should render ApplicantUserInfo correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
