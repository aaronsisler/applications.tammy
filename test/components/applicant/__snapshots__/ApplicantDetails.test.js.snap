// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicantDetails when applicant is defined when user contact info is NOT provided should render ApplicantUserInfo correctly 1`] = `
<div
  className="inbox_details"
>
  <div
    className="inbox_details_header"
  >
    <div
      className="inbox_details_header__actions"
    >
      <button
        className="inbox_details_header__mobile_button"
        onClick={[Function]}
      >
        Back to List
      </button>
    </div>
    <div
      className="inbox_details_header__content"
    >
      <div
        className="inbox_details_header__display_name"
      >
        Johnny Appleseed
      </div>
      <div
        className="inbox_details_header__phone_number"
      >
        <strong>
          Phone:
        </strong>
         
        Not Provided
      </div>
      <div
        className="inbox_details_header__email"
      >
        <strong>
          Email:
        </strong>
         
        Not Provided
      </div>
    </div>
  </div>
  <div
    className="inbox_details_content"
  >
    <ApplicantDetailsContent
      applicant={
        Object {
          "applicantId": "firstApplicantId",
          "applicantNotes": Array [
            Object {
              "noteMessage": "First note",
            },
            Object {
              "noteMessage": "Second note",
            },
          ],
          "applicantStatus": "APPLIED",
          "positionId": "firstPositionId",
          "user": Object {
            "addressLine1": "123 Evergreen Terrace",
            "addressLine2": "Apt. 456",
            "city": "Durham",
            "displayName": "Johnny Appleseed",
            "displayPhoneNumber": undefined,
            "email": undefined,
            "firstName": "Johnny",
            "lastName": "Appleseed",
            "middleName": "Middleton",
            "phoneAreaCode": "919",
            "phoneExt": "12345",
            "phoneLineNumber": "4567",
            "phonePrefix": "123",
            "postalCode": "27540",
            "state": "NC",
          },
          "userDocuments": Array [
            Object {
              "dateUploaded": "2018/01/01 01:23:45A",
              "documentName": "Document Name 1",
              "downloadUrl": "downloadUrl1",
              "userDocumentId": "firstDocumentId",
            },
            Object {
              "dateUploaded": "2018/02/02 01:23:45A",
              "documentName": "Document Name 2",
              "downloadUrl": "downloadUrl2",
              "userDocumentId": "secondDocumentId",
            },
          ],
        }
      }
    />
  </div>
</div>
`;

exports[`ApplicantDetails when applicant is defined when user contact info is provided should render ApplicantUserInfo correctly 1`] = `
<div
  className="inbox_details"
>
  <div
    className="inbox_details_header"
  >
    <div
      className="inbox_details_header__actions"
    >
      <button
        className="inbox_details_header__mobile_button"
        onClick={[Function]}
      >
        Back to List
      </button>
    </div>
    <div
      className="inbox_details_header__content"
    >
      <div
        className="inbox_details_header__display_name"
      >
        Johnny Appleseed
      </div>
      <div
        className="inbox_details_header__phone_number"
      >
        <strong>
          Phone:
        </strong>
         
        919-123-4567 ext. 12345
      </div>
      <div
        className="inbox_details_header__email"
      >
        <strong>
          Email:
        </strong>
         
        johnny.appleseed@gmail.com
      </div>
    </div>
  </div>
  <div
    className="inbox_details_content"
  >
    <ApplicantDetailsContent
      applicant={
        Object {
          "applicantId": "firstApplicantId",
          "applicantNotes": Array [
            Object {
              "noteMessage": "First note",
            },
            Object {
              "noteMessage": "Second note",
            },
          ],
          "applicantStatus": "APPLIED",
          "positionId": "firstPositionId",
          "user": Object {
            "addressLine1": "123 Evergreen Terrace",
            "addressLine2": "Apt. 456",
            "city": "Durham",
            "displayName": "Johnny Appleseed",
            "displayPhoneNumber": "919-123-4567 ext. 12345",
            "email": "johnny.appleseed@gmail.com",
            "firstName": "Johnny",
            "lastName": "Appleseed",
            "middleName": "Middleton",
            "phoneAreaCode": "919",
            "phoneExt": "12345",
            "phoneLineNumber": "4567",
            "phonePrefix": "123",
            "postalCode": "27540",
            "state": "NC",
          },
          "userDocuments": Array [
            Object {
              "dateUploaded": "2018/01/01 01:23:45A",
              "documentName": "Document Name 1",
              "downloadUrl": "downloadUrl1",
              "userDocumentId": "firstDocumentId",
            },
            Object {
              "dateUploaded": "2018/02/02 01:23:45A",
              "documentName": "Document Name 2",
              "downloadUrl": "downloadUrl2",
              "userDocumentId": "secondDocumentId",
            },
          ],
        }
      }
    />
  </div>
</div>
`;

exports[`ApplicantDetails when applicant is undefined should render ApplicantUserInfo correctly 1`] = `
<div
  className="inbox_details__empty inbox_mobile"
>
  Please select an applicant to view
</div>
`;
